src/bitcoin/wallet.rs

```diff
@@ -1,27 +1,33 @@
-use crate::{
-    bitcoin::{Address, Amount, Client, Network, WalletInfoResponse},
-    seed::Seed,
+use crate::bitcoin::{Address, Amount, Client, Network, WalletInfoResponse};
+use crate::seed::Seed;
+use ::bitcoin::{
+    hash_types::PubkeyHash,
+    hashes::Hash,
+    hashes::{sha512, HashEngine, Hmac, HmacEngine},
+    secp256k1,
+    secp256k1::SecretKey,
+    util::bip32::ExtendedPrivKey,
+    PrivateKey, Transaction, Txid,
 };
 use anyhow::Context;
-use bitcoin::{
-    hash_types::PubkeyHash, hashes::Hash, secp256k1::SecretKey, PrivateKey, Transaction, Txid,
-};
+use bitcoin::util::bip32::ChainCode;
 use rand::RngCore;
 use sha2::{Digest, Sha256};
+use std::path::Path;
 use url::Url;
 
 #[derive(Debug, Clone)]
 pub struct Wallet {
     /// The wallet is named `nectar_x` with `x` being the first 4 byte of the public key hash
     name: String,
     bitcoind_client: Client,
-    private_key: bitcoin::PrivateKey,
+    seed: Seed,
     network: Network,
 }
 
 impl Wallet {
     pub async fn new(seed: Seed, url: Url, network: Network) -> anyhow::Result<Wallet> {
-        let key = seed.secret_key()?;
+        let key = secp256k1::SecretKey::from_slice(&seed.bytes())?;
 
         let private_key = ::bitcoin::PrivateKey {
```

Comment:
:wrench: Now that the wallet just holds the seed, it's really weird to generate a private key just for the purpose of generating a name for the wallet.